diff --git a/builder/comp-builder.nix b/builder/comp-builder.nix
index 7f7fe586..9408f191 100644
--- a/builder/comp-builder.nix
+++ b/builder/comp-builder.nix
@@ -511,6 +511,7 @@ let
         mkdir -p $out/bin
         if [ -f ${testExecutable} ]; then
           mkdir -p $(dirname $out/bin/${exeName})
+          ${lib.optionalString stdenv.buildPlatform.isLinux "sync"}
           ${if stdenv.hostPlatform.isGhcjs then ''
             cat <(echo \#!${lib.getBin buildPackages.nodejs-12_x}/bin/node) ${testExecutable} >| $out/bin/${exeName}
             chmod +x $out/bin/${exeName}
